platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -956,14 +1036,29 @@ int _odp_ipsec_sa_replay_update(ipsec_sa_t *ipsec_sa, uint32_t seq,
 		}
 
 		new_state = (((uint64_t)mask) << 32) | max_seq;
-
-		cas = odp_atomic_cas_acq_rel_u64(&ipsec_sa->hot.in.antireplay,
+		cas = odp_atomic_cas_acq_rel_u64(&ipsec_sa->hot.in.wintop_seq,
 						 &state, new_state);
 	}
-
 	return 0;
 }
 
+int _odp_ipsec_sa_replay_update(ipsec_sa_t *ipsec_sa, uint32_t seq,
+				odp_ipsec_op_status_t *status)
+{
+	int ret;
+
+	if (!ipsec_sa->antireplay)
+		return 0;
```

Comment:
This check is unnecessary. It has already been done by the caller.

Reply:
Will check and remove it.